### DISCUSSION TITLE
How to have custom key bind action to grab contents of preview window and show new content in preview window.

### DISCUSSION THREAD

**Discussion by asknet** [ibhagwan/fzf-lua#1309] [Q&A]
@ibhagwan 
My requirement is on the execution of a custom key bind action that executes something and puts some content back into the preview window. If possible grab the content of the preview window before executing this key bind. fzf have something like this. Read fzf-lua docs, still unable to figure out how to do this. Appreciate the help

`fzf --preview 'cat {}' --bind 'ctrl-d:preview:file {}' --bind 'ctrl-f:preview:cat {}'`

**Comment by ibhagwan** [comment]
> fzf have something like this. Read fzf-lua docs, still unable to figure out how to do this

If it’s possible to do with fzf it’s possible to do in fzf-lua, you can send raw args to fzf using `fzf_cli_args = …`, if you wanna use `{}` make sure to disable all icons or use a combination of delimiter and a specific field, I.e. `{2}`.

P.S. I don’t understand what you’re trying to do or what’s the use case, if you can clarify properly I’m sure there’s a more efficient way to achieve what you want with fzf-lua.

**Reply by ibhagwan** [reply]
Btw, here’s how to replicate your exact command with fzf-lua:
```lua
require('fzf-lua').files({ git_icons= false, file_icons=false, previewer=false, preview='cat {}', fzf_cli_args=[[--bind 'ctrl-d:preview:file {}' --bind 'ctrl-f:preview:cat {}']] })
```

**Reply by asknet** [reply]
Thanks for the quick response 🙏🏽

Instead of sending as CLI args, can this be achieved using actions['ctrl-d'] = function (s) 
-- do something 
-- output in existing preview window 
done

**Reply by ibhagwan** [reply]
You can use an action like this:
```lua
  actions = {
    ['ctrl-d'] = { fn = function(sel) … end, exec_silent=true, postfix = 'preview:cat {}' },
  }
```

**Reply by asknet** [reply]
Thanks! 
In the above example, Postfix is acting on selected iteym. What I'm trying to do is I take the selected item, do some transformation,  invoke rest API and show the output of rest API in preview window. Can all this be done with lua action callback? 

I'm thankful for your time on this matter

**Reply by ibhagwan** [reply]
You can but the code will have to be a bit different, you convert a lua function to a preview command using `raw_preview_action_cmd`:
https://github.com/ibhagwan/fzf-lua/blob/41ea49cfc1b3f0586753bbeb7850474f28dc22cf/lua/fzf-lua/shell.lua#L153

The return value will be a shell command you can use with the preview:
```lua
  -- second arg is the fzf field index identifier (can be `{+}` for multi select)
  -- third arg is the debug flag (will output debug data in the preview win)
  -- change to `false` or `nil` when done
  local act_cmd = shell.raw_preview_action_cmd(fn(item) … end, '{}', true)
  opts.fzf_cli_args = [[--bind 'ctrl-f:preview:]] .. act_cmd .. [[']]
```

**Reply by asknet** [reply]
Thank you. I will explore these suggestions. I'm pretty new to lua as well and I'm taking some time to understand some of these things.
